internal/fwschema/nested_attribute.go

```diff
@@ -17,3 +17,18 @@ type NestedAttribute interface {
 	// does not represent nested attributes.
 	GetNestingMode() NestingMode
 }
+
+// NestedAttributesEqual is a helper function to perform equality testing on two
+// NestedAttribute. NestedAttribute Equal implementations should still compare
+// the concrete types in addition to using this helper.
+func NestedAttributesEqual(a, b NestedAttribute) bool {
+	if a.GetNestingMode() != b.GetNestingMode() {
+		return false
+	}
+
+	if !a.GetNestedObject().Equal(b.GetNestedObject()) {
+		return false
+	}
+
+	return AttributesEqual(a, b)
```

Comment:
Is there a reason that we first compare the nested objects before we compare the attributes? If I'm understanding the `BlocksEqual` function above it is the other way around (I don't think it matters either way, just curious)

Reply:
Nope and in fact, maybe we should compare the "easier" bits first to save a little on performance. Would you want to submit a followup PR to adjust these? (I don't think they need a benchmark test or anything)

Reply:
sounds good, I have no problem submitting the followup PR 👍🏻